test/multi.html

```diff
@@ -103,31 +103,106 @@
         assert.equal(s.selectedItems[1], s.children[2]);
       });
 
-      test('fire iron-select/deselect events', function() {
-        // setup listener for iron-select event
-        var selectEventCounter = 0;
+      test('fire iron-select/deselect events when selectedValues changes', function() {
+        // setup listener for iron-select/deselect events
+        var items = [s.children[0], s.children[1], s.children[2]], 
+            selectEventCounters = [0, 0, 0], 
+            deselectEventCounters = [0, 0, 0];
+
         s.addEventListener('iron-select', function(e) {
-          selectEventCounter++;
+          selectEventCounters[items.indexOf(e.detail.item)]++;
         });
-        // setup listener for core-deselect event
-        var deselectEventCounter = 0;
         s.addEventListener('iron-deselect', function(e) {
-          deselectEventCounter++;
+          deselectEventCounters[items.indexOf(e.detail.item)]++;
         });
-        // tap to select an item
-        s.children[0].dispatchEvent(new CustomEvent('tap', {bubbles: true}));
+
+        // programatically select values 0 and 1 (both fire select)
+        s.selectedValues = [0, 1];
+        
+        // programatically select values 1 and 2 (2 fires select, 0 fires deselect)
+        s.selectedValues = [1, 2];
+
+        // programatically deselect all values (1 and 2 fire deselect)
+        s.selectedValues = [];
+
         // check events
-        assert.equal(selectEventCounter, 1);
-        assert.equal(deselectEventCounter, 0);
-        // tap on already selected item should deselect it
-        s.children[0].dispatchEvent(new CustomEvent('tap', {bubbles: true}));
-        // check selectedValues
-        assert.equal(s.selectedValues.length, 0);
-        // check class
-        assert.isFalse(s.children[0].classList.contains('iron-selected'));
+        assert.equal(selectEventCounters[0], 1);
+        assert.equal(deselectEventCounters[0], 1);
+        assert.equal(selectEventCounters[1], 1);
+        assert.equal(deselectEventCounters[1], 1);
+        assert.equal(selectEventCounters[2], 1);
+        assert.equal(deselectEventCounters[2], 1);
+      });
+
+      test('fire iron-select/deselect events when toggling items', function() {
+        // setup listener for iron-select/deselect events
+        var items = [s.children[0], s.children[1], s.children[2]], 
+            selectEventCounters = [0, 0, 0], 
+            deselectEventCounters = [0, 0, 0];
+
+        s.addEventListener('iron-select', function(e) {
+          selectEventCounters[items.indexOf(e.detail.item)]++;
+        });
+        s.addEventListener('iron-deselect', function(e) {
+          deselectEventCounters[items.indexOf(e.detail.item)]++;
+        });
+
+        // tap to select items 0 and 1 (both fire select)
+        items[0].dispatchEvent(new CustomEvent('tap', {bubbles: true}));
```

Comment:
We've got a thing for this: https://github.com/PolymerElements/iron-test-helpers/blob/66ad13aebf0dd4a58027175170af1f58a2f9fc36/mock-interactions.js#L174

It's already imported, so you can just do `MockInteractions.tap(items[0]);`.